front/app/utils/fileUtils.ts

```diff
@@ -61,16 +62,16 @@ export async function convertUrlToUploadFile(
       response.blob()
     );
     const urlFilename = url.substring(url.lastIndexOf('/') + 1);
-    const uploadFile = convertBlobToFile(
-      blob,
-      filename || urlFilename
-    ) as UploadFile;
-    const base64 = await getBase64FromFile(uploadFile);
-    uploadFile.url = url;
-    uploadFile.base64 = base64;
-    uploadFile.remote = true;
-    uploadFile.filename = filename || urlFilename;
-    uploadFile.id = id || undefined;
+    const file = convertBlobToFile(blob, filename || urlFilename);
+    const base64 = await getBase64FromFile(file);
+    const uploadFile: UploadFile = Object.assign(file, {
+      url,
+      base64,
+      remote: true,
+      filename: filename || urlFilename,
+      id: fileId || undefined,
+    });
+
```

Comment:
Prefer this: creating a file and not calling its type UploadFile via type casting and adding props later, but give the type later (as I do with const uploadFile: UploadFile)  so you have to make sure all necessary properties are there.